lib/bolt/util.rb

```diff
@@ -169,12 +169,13 @@ def postwalk_vals(data, skip_top = false, &block)
       # references to the same object and prevents endless recursion.
       # Credit to Jan Molic via https://github.com/rubyworks/facets/blob/master/LICENSE.txt
       def deep_clone(obj, cloned = {})
-        cloned[obj.object_id] if cloned.include?(obj.object_id)
+        return cloned[obj.object_id] if cloned.include?(obj.object_id)
 
         begin
           cl = obj.clone
-        rescue TypeError
-          # unclonable (TrueClass, Fixnum, ...)
+        rescue StandardError
```

Comment:
`StandardError` is more broad than I'm particularly comfortable with - do you have notes on the error types you were seeing with Java objects so we could try to narrow this down?

Reply:
In addition to `TypeError` I was seeing specifically `Encoding::InvalidByteSequenceError`, rescuing that or any `EncodingError` would still not stop the problem. When I tried to figure out what the error was by inspecting the error object (excluding TypeError and EncodingError) I was getting `exception class/object expected` error messages. When I hit that I moved on. I can revisit it.

Reply:
InvalidByteSequence sounds like maybe there's something wrong somewhere else and we're just seeing it in this functon? Perhaps a Java UTF-16 vs Ruby UTF-8 issue?

Reply:
I think there might be a problem with a java object not having a `clone` method:
```
If JRuby cannot find a matching method, it tries to pass the actual JRuby objects instead (that is, the Java objects from the JRuby implementation). A consequence of this is that if this fails you will see an error message stating that JRuby hasn't found a method taking an object of class org.jruby.RubyObject instead of the actual type.
```
^^ from https://github.com/jruby/jruby/wiki/CallingJavaFromJRuby

Reply:
Hm, the `EncodingError` does not seem to be an issue when I simply rescue the `Java::JavaLang::CloneNotSupportedException`.

Reply:
I would catch the CloneNotSupportedException, or you can do a `instanceof` conditional for the `java.Lang.Cloneable` interface. In Java:

```
if (obj instanceof Cloneable) {
  cl = obj.clone();
}
```